streams/src/main/java/org/apache/kafka/streams/internals/InternalStreamsConfig.java

```diff
@@ -0,0 +1,230 @@
+import org.apache.kafka.streams.errors.DeserializationExceptionHandler;
+import org.apache.kafka.streams.errors.ProductionExceptionHandler;
+import org.apache.kafka.streams.processor.TimestampExtractor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class InternalStreamsConfig extends StreamsConfig {
+    private static final Logger log = LoggerFactory.getLogger(StreamsConfig.class);
+
+    // This is settable in the main Streams config, but it's a private API for now
+    public static final String TASK_ASSIGNOR_CLASS = "__internal.task.assignor.class__";
+    // These are not settable in the main Streams config; they are set by the StreamThread to pass internal
+    // state into the assignor.
+    public static final String REFERENCE_CONTAINER_PARTITION_ASSIGNOR = "__reference.container.instance__";
+    // This is settable in the main Streams config, but it's a private API for testing
+    public static final String ASSIGNMENT_LISTENER = "__assignment.listener__";
+    // Private API used to control the emit latency for left/outer join results (https://issues.apache.org/jira/browse/KAFKA-10847)
+    public static final String EMIT_INTERVAL_MS_KSTREAMS_OUTER_JOIN_SPURIOUS_RESULTS_FIX = "__emit.interval.ms.kstreams.outer.join.spurious.results.fix__";
+    // Private API used to control the emit latency for windowed aggregation results for ON_WINDOW_CLOSE emit strategy
+    public static final String EMIT_INTERVAL_MS_KSTREAMS_WINDOWED_AGGREGATION = "__emit.interval.ms.kstreams.windowed.aggregation__";
+    // Private API used to control the usage of consistency offset vectors
+    public static final String IQ_CONSISTENCY_OFFSET_VECTOR_ENABLED = "__iq.consistency.offset.vector.enabled__";
+    // Private API used to control the prefix of the auto created topics
+    public static final String TOPIC_PREFIX_ALTERNATIVE = "__internal.override.topic.prefix__";
+    // Private API to enable the state updater (i.e. state updating on a dedicated thread)
+    public static final String STATE_UPDATER_ENABLED = "__state.updater.enabled__";
+
+
+
+    public InternalStreamsConfig(final StreamsConfig streamsConfig) {
+        super(streamsConfig.originals(), false);
+    }
+
+    public InternalStreamsConfig(final Map<?, ?> configs) {
+        super(configs, false);
+    }
+
+
+
+    /**
+     * Return a copy of the config definition.
+     *
+     * @return a copy of the config definition
+     */
+    @SuppressWarnings({"deprecation", "unused"})
+    public static ConfigDef configDef() {
+        return new ConfigDef(CONFIG);
+    }
+
+    public static boolean stateUpdaterEnabled(final Map<String, Object> configs) {
+        return getBoolean(configs, STATE_UPDATER_ENABLED, true);
+    }
+
+    public static boolean getBoolean(final Map<String, Object> configs, final String key, final boolean defaultValue) {
```

Comment:
Kept the `get` here and below for similar oned, because `AbstractConfig` also uses `get` -- not 100% sure what is cleaner.